packages/bpk-component-slider/src/BpkSlider.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import Slider from 'react-slider';
+import { cssModules } from 'bpk-react-utils';
+import React, { Component } from 'react';
+
+import STYLES from './bpk-slider.scss';
+
+const getClassName = cssModules(STYLES);
+
+class BpkSlider extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { value: props.value };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(value) {
+    this.setState({ value });
+    this.props.onChange(value);
+  }
+
+  render() {
+    const {
+      min,
+      max,
+      step,
+      large,
+      valueTemplate,
+      minDistance,
+      className,
+      onAfterChange,
+    } = this.props;
+
+    const sliderClass = [];
+    const handleClass = [];
+    const barClass = [];
+
+    sliderClass.push(getClassName('bpk-slider'));
+    sliderClass.push(getClassName(this.state.value.length
+      ? 'bpk-slider__range'
+      : 'bpk-slider__simple'));
+    handleClass.push(getClassName('bpk-slider-handle'));
+    barClass.push(getClassName('bpk-slider-bar'));
```

Comment:
This classnames logic can be simplified to something along the lines of this:
```
    const sliderClass = [getClassName('bpk-slider')];
    const handleClass = [getClassName('bpk-slider-handle')];

    sliderClass.push(getClassName(this.state.value.length ? 'bpk-slider__range' : 'bpk-slider__simple'));
    if (large) handleClass.push(getClassName('bpk-slider-handle--large'));
    if (className) sliderClass.push(getClassName(className));
```

In short... 
```
const myArray = [];
myArray.push(item)
```
can be:
```
const myArray = [item];
```

etc.